tests/lib/devicetree/src/main.c

```diff
@@ -200,6 +200,27 @@ static void test_has_compat(void)
 	zassert_equal(compats, 0x3, "as bit array");
 }
 
+static void test_has_status(void)
+{
+	zassert_equal(DT_NODE_HAS_STATUS(DT_NODELABEL(test_gpio_1), okay),
+		      1, "vnd,gpio okay");
+	zassert_equal(DT_NODE_HAS_STATUS(DT_NODELABEL(test_gpio_1), disabled),
+		      0, "vnd,gpio not disabled");
+
+	zassert_equal(DT_NODE_HAS_STATUS(DT_NODELABEL(test_no_status), okay),
+		      1, "vnd,gpio okay");
+	zassert_equal(DT_NODE_HAS_STATUS(DT_NODELABEL(test_no_status), disabled),
+		      0, "vnd,gpio not disabled");
+
+/* Disabled for now until we generate something for non-"enabled" nodes */
+#if 0
```

Comment:
Leaving a comment here since we've merged ifdeffed out tests before, to make sure it doesn't happen this time